design/m1_push_to_file_design.md

```diff
@@ -0,0 +1,1163 @@
+from files at expected locations, further minimizing changes to the
+application.
+
+The M1 Push-to-File feature makes use of
+[Kubernetes Pod Annotations](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/)
+to provide a method of flexibly configuring the mapping of secrets to desired
+target locations.
+
+The M1 Push-to-File feature also adds support for configuring other
+Secrets Provider container/application settings using Kubernetes annotations
+as an alternative to using Pod environment variable settings. Allowing
+Secrets Provider to be more generally configured via Pod annotations
+provides a uniform mechanism for configuring the Secrets Provider, and should
+lend itself to an easier upgrade/migration to the proposed CyberArk Dynamic
+Sidecar Injector design.
+
+## Solution
+
+### User Experience
+
+#### Annotations for Secrets Provider Container/Application Configuration
+
+The Secrets Provider container/application configuration can be set using
+either Pod annotations or by using Pod environment variable settings
+([legacy configuration](https://docs.cyberark.com/Product-Doc/OnlineHelp/AAM-DAP/11.2/en/Content/Integrations/Kubernetes_deployApplicationsConjur-k8s-Secrets.htm#Step4AddtheCyberArkSecretsProviderforKubernetesimagetotheapplicationdeploymentmanifest)).
+If both a Pod Annotation and a Pod environment variable is configured for
+any particular Secrets Provider setting, then the Pod Annotation configuration
+takes precedence.
+
+_**QUESTION: Should we deprecate the current configuration method that uses
+   environment variable settings?**_
+
+Note that Pod Annotation configuration will not be supported for the
+`MY_POD_NAME` and `MY_POD_NAMESPACE` settings. These settings will continue
+to use the standard
+[Kubernetes Downward API configuration](https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/).
+
+The following annotations are supported for Secrets Provider configuration.
+Please refer to the
+[Secrets Provider documentation](https://docs.cyberark.com/Product-Doc/OnlineHelp/AAM-DAP/11.2/en/Content/Integrations/Kubernetes_deployApplicationsConjur-k8s-Secrets.htm#Step4AddtheCyberArkSecretsProviderforKubernetesimagetotheapplicationdeploymentmanifest)
+for a description of each setting:
+
+| Annotation | Equivalent<br /> Environment Variable | Description, Notes |
+|--------------------------------------|---------------------|----------------------------------|
+| conjur.org/<br />authn-identity      | CONJUR_AUTHN_LOGIN  | Required value. Example: `host/conjur/authn-k8s/cluster/apps/inventory-api` |
+| conjur.org/<br />container-mode      | CONTAINER_MODE      | Allowed values: <ul><li>`init`</li><li>`application`</li><li>`sidecar`</li></ul>Defaults to `sidecar` |
+| conjur.org/<br />secrets-destination | SECRETS_DESTINATION | Required value. Allowed values: <ul><li>`file`</li><li>`k8s_secret`</li></ul> |
```

Comment:
~maybe highlight that "file" is to be added here? at first, this chart read like a list of _current_ supported config options (ie it doesn't include the new file-based annotations), so I was surprised to see this option here~

~alternatively, we could add all of the new options and just highlight the ones that are new / to be added in this effort~

editing to just say instead: maybe we could highlight the new options in these tables? e.g. "New configuration settings or options are highlighted in **bold**" and then make "file" and `conjur.org/conjur-configmap` bold